app/src/pages/Lobby/WaitingRoom.tsx

```diff
@@ -26,9 +36,12 @@ function WaitingRoom() {
   const canStartGame =
     canSeeStartGameButton &&
     currentGame.seconds_per_turn &&
-    currentGame.num_entries_per_player &&
+    find(playersWithUsernames, (player) => player.id === currentPlayer.id) &&
     playersWithUsernames.length >= MIN_NUMBER_OF_PLAYERS &&
-    find(playersWithUsernames, (player) => player.id === currentPlayer.id)
+    ((props.cardPlayStyle === GameCardPlayStyleEnum.PlayersSubmitWords &&
+      currentGame.num_entries_per_player) ||
+      (props.cardPlayStyle === GameCardPlayStyleEnum.HostProvidesWords &&
+        Boolean(props.wordList)))
```

Comment:
more sophisticated "can start game" logic - to prevent starting the game w/an empty list of words if host is supposed to be providing them.